### PR TITLE
Enhance Data Authorization Handler

### PR DESCRIPTION
**Pull request by AyakorK**
Hello ! I'm doing this pull request following changes that were requested on my authorization handler.

Following this I removed the structure that was judged not useful, restored GDPR, added plenty of things like Zipcode, City, minimum_age and an help text that precise to the user that he needs to complete this authorization to post some proposals.

If there are some changes that you want me to do don't mind bothering me ! I'll be there for it.

This PR is linked to #64 , commits above [9795e72](https://github.com/OpenSourcePolitics/decidim-cd44/pull/69/commits/9795e7237a60f3884bd7b45fa17d4a8e09f50f4e) are outdated but still, as I started for my initial branch, are still pushed, but changes that are coming from it were already merged and conflicts that were linked are now resolved.

### CODE REVIEW

File: app/services/decidim/data_authorization_handler.rb [hunk 1]
```diff
@@ -6,16 +6,29 @@ class DataAuthorizationHandler < Decidim::AuthorizationHandler
...
+    attribute :city, String
+
 
     validates :firstname, presence: true, format: { with: /\A[a-zA-ZÀ-ÿ'\-\s]+\z/,
                                                     message: I18n.t("decidim.authorization_handlers.data_authorization_handler.errors.error_message") }
     validates :lastname, presence: true, format: { with: /\A[a-zA-ZÀ-ÿ'\-\s]+\z/,
                                                    message: I18n.t("decidim.authorization_handlers.data_authorization_handler.errors.error_message") }
     validates :phone, presence: true
 
+    validates :gdpr, acceptance: { message: I18n.t("decidim.authorization_handlers.data_authorization_handler.errors.acceptance_error") }
```
Comment (Quentinchampenois) [inline app/services/decidim/data_authorization_handler.rb:21]: Why not using the default devise message ?

File: app/services/decidim/data_authorization_handler.rb [hunk 2]
```diff
@@ -6,16 +6,29 @@ class DataAuthorizationHandler < Decidim::AuthorizationHandler
...
+    validates :minimum_age, acceptance: { message: I18n.t("decidim.authorization_handlers.data_authorization_handler.errors.acceptance_error") }
+
+    validates :zipcode, presence: true, format: { with: /\A[0-9]{5}\z/,
+                                                  message: I18n.t("decidim.authorization_handlers.data_authorization_handler.errors.error_message") }
+
+    validates :city, presence: { message: I18n.t("decidim.authorization_handlers.data_authorization_handler.errors.no_zipcode_result") }
+
     def metadata
-      super.merge(firstname: firstname, lastname: lastname, phone: phone, structure: structure)
+      super.merge(firstname: firstname, lastname: lastname, phone: phone, zipcode: zipcode, city: city)
```
Comment (Quentinchampenois) [inline app/services/decidim/data_authorization_handler.rb:31]: I think we should save `gdpr` and `minimum_age` acceptance, what do you think @OuardaOUBAHRI ?

File: config/locales/en.yml [hunk 3]
```diff
@@ -39,18 +39,26 @@ en:
     authorization_handlers:
       data_authorization_handler:
         errors:
+          acceptance_error: You must check this box to continue
```
Comment (Quentinchampenois) [inline config/locales/en.yml:42]: Not sure this translation is necessary

**Review by Quentinchampenois — CHANGES_REQUESTED** [review]
Thank you, I notice some extra translation not necessary to me, otherwise it looks great

File: app/services/decidim/data_authorization_handler.rb (hunk 2 above)
Comment (OuardaOUBAHRI) [reply app/services/decidim/data_authorization_handler.rb:31]: Indeed, we need these information. They should be saved too

**Review by OuardaOUBAHRI — CHANGES_REQUESTED** [review]
Thank you for the work ! 
I have some recommandations, find them on [Notion](https://www.notion.so/opensourcepolitics/CD44-D-veloppement-d-une-autorisation-de-r-colte-de-donn-es-6755ff7bde06464a9ea4f4ef722c9d20#fd1b51931a1b44e3b6b160060619b0b8)

**Review by OuardaOUBAHRI — APPROVED** [review]
All good for me ! Thank you 💛

**Review by paulinebessoles — APPROVED** [review]

**Pull request merged**